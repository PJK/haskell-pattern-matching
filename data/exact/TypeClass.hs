module TypeClass where

unsound :: Num a => a -> Int
unsound x
    | x <= 0 = 1
    | x >= 1 = 2
    -- Exhaustive for integrals, but not for double, for example.
    -- Our solver says nothing while this should probably get some warning that it's not exhaustive.
