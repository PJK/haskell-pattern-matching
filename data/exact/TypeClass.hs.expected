{
    "tag": "AnalysisSuccess",
    "contents": [
        [],
        [
            [
                "unsound",
                [
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            }
                        ],
                        [
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            }
                        ]
                    ],
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            }
                        ],
                        [
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            }
                        ]
                    ]
                ]
            ]
        ]
    ]
}
